azavea/openstreetmap-website
# Feature/kak/vagrant setup fixes

## Overview

Modifies the existing vagrant setup to get to a functional OSM web server with a data import.


## Demo

Viewing a street segment in the test area, it looks like the street information mostly came as a bulk import from PENNDOT/Census Tiger data, and lacks `sidewalk:` tags to indicate if the roadway has sidewalks. It also looks like sidewalks have not been added as separate ways. This should make it easier to import the [sidewalk data from DVRPC](https://dvrpc-dvrpcgis.opendata.arcgis.com/datasets/dvrpc-pedestrian-network-sidewalks) later as separate ways, since it hopefully won't be necessary to merge the new sidewalk ways with existing features, or at least not many of them.

![image](https://user-images.githubusercontent.com/960264/70730820-a9c29400-1cd3-11ea-81f3-73828a985710.png)


## Notes

For test data, I downloaded an `.osm` file using the Overpass API export option [here](https://www.openstreetmap.org/export#map=13/40.0144/-75.2766), which is for bounds that approximately cover Ardmore.

For larger areas, OSM data with edits can be downloaded from the [internal geofabrik](https://osm-internal.download.geofabrik.de/) site; the default downloads that do not require login do not contain edits.

I named the new scripts folder I added `strta` and not `scripts`, since there is already a `script` directory in the project.

The changes to `db/structure.sql` were autogenerated by changes to the provisioning process that allow for `osmosis` imports to work, and so can be ignored.

The `osmosis` import tool uses the APIDB schema format shared by the OSM website project here, but this project has more recent migrations that are not in [the osmosis migrations list](https://github.com/openstreetmap/osmosis/blob/2219470cef1f73f5d1319c57149c84b398e767ce/osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbVersionConstants.java), which has not had its migration version numbers updated since 2013. The official OSM website project documentation suggests [here](https://github.com/openstreetmap/openstreetmap-website/blob/master/CONFIGURE.md#populating-the-database) simply running osmosis with a flag to ignore schema version mismatches, which results in errors regarding missing fields. Instead, I've modified the provisioning script to first apply database migrations up to the last `osmosis` migration from 2013, then run the `osmosis` import, then finally apply the remaining website migrations to the filled database.


## Testing

 - Download an OSM extract that contains user and edit data and put it in `data`
 - Run `vagrant up`
 - Provisioning should succeed
 - Run `./srta/serve.sh` to start the web server
 - Web server should be accessible at http://localhost:3003/
 - Create a user account and iD editor OAuth app following the steps in `AZAVEA_README.md`
 - OSM features should load for the OSM extract area in the iD editor (but slowly)


Connects azavea/geospatial-apps-tasks#43
